generator/templates/structs.hpp.j2

```diff
@@ -0,0 +1,126 @@
+{%- macro struct_declarations(bus, msg, msg_name=msg.name) -%}
+{%- if msg.frame is defined %}
+    namespace {{msg_name}} {
+        const uint32_t key = 0x{{ '%0X' % msg.key }};
+        {% for sub_frame in msg.frame %}
+            {{ struct_declarations(bus, sub_frame) }}
+        {% endfor %}
+    } // {{msg_name}}
+
+{% else %}
+    struct {{msg_name}}_T : public Message_T {
```

Comment:
A static function to get the input struct might be nice

Reply:
Input struct is in messages, in particular one way to easily get it is `messages[static_cast<int> (Message_Type_T::J_AA)]`